dashboard/app/controllers/omniauth_callbacks_controller.rb

```
@@ -5,6 +5,46 @@ class OmniauthCallbacksController < Devise::OmniauthCallbacksController
 
   # GET /users/auth/:provider/callback
   def all
+    auth_params = request.env['omniauth.params']
+    if auth_params['state'] == 'connect_provider' && current_user
```

If the `'connect_provider'` state is set but there's no `current_user` do we really want to fall through to the login case?  I think we might prefer to fail hard here, maybe with a 401 or 403, rather than have surprising (but "successful") behavior.